cromwell_tools/_cromwell_api.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime, timedelta
+import time
+
+import requests
+import requests_mock
+
+from tenacity import retry, wait_exponential, stop_after_delay
+from ._utilities import validate_cromwell_label, prepare_workflow_manifest
+
+
+class WorkflowFailedException(Exception):
+    pass
+
+
+class WorkflowUnknownException(Exception):
+    pass
+
+
+# todo functools partial for get, post (set the authenticate commands)
+class CromwellAPI:
+
+    # todo remove this if it is not critical for mocking
+    requests = requests
+
+    _status_endpoint = '/api/workflows/v1/{uuid}/status'
+    _workflow_endpoint = '/api/workflows/v1'
+    _metadata_endpoint = '/api/workflows/v1/{uuid}/metadata'
+    _health_endpoint = '/api/engine/v1'
```

Comment:
we should probably use the status endpoint here

Reply:
Yep, would be very excited to move to those, but I didn't want to change too much in one PR, so I kept what was in the repo already.

Reply:
I don't understand this comment.